src/constants/volumeRayMarchShader.ts

```diff
@@ -1,297 +1,293 @@
+  float tdist = 0.0;
+  int numSteps = 0;
+  vec4 pos, col;
   // We need to be able to scale the alpha contrib with number of ray steps,
   // in order to make the final color invariant to the step size(?)
   // use maxSteps (a constant) as the numerator... Not sure if this is sound.
-  "    float s = 0.5 * float(maxSteps) / csteps;",
-  "for(int i=0; i<maxSteps; i++){",
-  "  pos = eye_o + eye_d*t;",
-  // !!! assume box bounds are -0.5 .. 0.5.  pos = (pos-min)/(max-min)
-  // scaling is handled by model transform and already accounted for before we get here.
-  // AABB clip is independent of this and is only used to determine tnear and tfar.
-  "  pos.xyz = (pos.xyz-(-0.5))/((0.5)-(-0.5)); //0.5 * (pos + 1.0); // map position from [boxMin, boxMax] to [0, 1] coordinates",
-  "  col = sampleStack(textureAtlas,pos);",
-
-  "      col.xyz *= BRIGHTNESS;",
-
-  "    if (maxProject != 0) {",
-  "      C = accumulateMax(col, s, C);",
-  "    } else {",
-  // for practical use the density only matters for regular volume integration
-  "      col.w *= DENSITY;",
-  "      C = accumulate(col, s, C);",
-  "    }",
-  "    t += tstep;",
-  "    numSteps = i;",
-
-  "    if (t  > tend || t > tbegin+clipFar ) break;",
-  "    if (C.w > 1.0 ) break;",
-  "}",
-
-  "  return C;",
-  "}",
-  "void main()",
-  "{",
-  "  gl_FragColor = vec4(0.0);",
-  "  vec2 vUv = gl_FragCoord.xy/iResolution.xy;",
-
-  "  vec3 eyeRay_o, eyeRay_d;",
-
-  "  if (isOrtho == 0.0) {",
-  // for perspective rays:
-  // world space camera coordinates
-  // transform to object space
-  "    eyeRay_o = (inverseModelViewMatrix * vec4(0.0, 0.0, 0.0, 1.0)).xyz;",
-  "    eyeRay_d = normalize(pObj - eyeRay_o);",
-  "  }",
-  "  else {",
-  // for ortho rays:
-  "    float zDist = 2.0;",
-  "    eyeRay_d = (inverseModelViewMatrix*vec4(0.0, 0.0, -zDist, 0.0)).xyz;",
-  "    vec4 ray_o = vec4(2.0*vUv - 1.0, 1.0, 1.0);",
-  "    ray_o.xy *= orthoScale;",
-  "    ray_o.x *= iResolution.x/iResolution.y;",
-  "    eyeRay_o   = (inverseModelViewMatrix*ray_o).xyz;",
-  "  }",
+  float s = 0.5 * float(maxSteps) / csteps;
+  for(int i=0; i<maxSteps; i++) {
+    pos = eye_o + eye_d*t;
+    // !!! assume box bounds are -0.5 .. 0.5.  pos = (pos-min)/(max-min)
+    // scaling is handled by model transform and already accounted for before we get here.
+    // AABB clip is independent of this and is only used to determine tnear and tfar.
+    pos.xyz = (pos.xyz-(-0.5))/((0.5)-(-0.5)); //0.5 * (pos + 1.0); // map position from [boxMin, boxMax] to [0, 1] coordinates
+
+    vec4 col = sampleAs3DTexture(textureAtlas, pos);
+    if (maxProject == 0) {
+      col = luma2Alpha(col, GAMMA_MIN, GAMMA_MAX, GAMMA_SCALE);
+    }
+
+    col.xyz *= BRIGHTNESS;
+
+    if (maxProject != 0) {
+      C = accumulateMax(col, s, C);
+    } else {
+      // for practical use the density only matters for regular volume integration
+      col.w *= DENSITY;
+      C = accumulate(col, s, C);
+    }
```

Comment:
small micro-optimization:  combine the if(maxProject==0) on line 217 with the if(maxProject!=0) statement on lines 223-229 so there is only a single if conditional. something like this?
```
if (maxProject == 0) {
      col = luma2Alpha(col, GAMMA_MIN, GAMMA_MAX, GAMMA_SCALE);
      col.xyz *= BRIGHTNESS;
      // for practical use the density only matters for regular volume integration
      col.w *= DENSITY;
      C = accumulate(col, s, C);
    }
    else {
      col.xyz *= BRIGHTNESS;
      C = accumulateMax(col, s, C);
    } 
```

Reply:
honestly i'm not even sure if that would have much perf impact so i leave it up to you

Reply:
yeah I realized I ought to do that as soon as I PR'ed - for code cleanliness as well as performance